editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+          }
+        }
+      });
+
+      // handle authentication for the service geodatabase
+      AuthenticationManager.setAuthenticationChallengeHandler(new DefaultAuthenticationChallengeHandler());
+
+      // create and load a service geodatabase
+      serviceGeodatabase = new ServiceGeodatabase("https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/DamageAssessment/FeatureServer");
+      serviceGeodatabase.loadAsync();
+      serviceGeodatabase.addDoneLoadingListener(() -> {
+        if (serviceGeodatabase.getLoadStatus() == LoadStatus.LOADED) {
+
+          // when the service geodatabase has loaded get the default version
+          defaultVersion = serviceGeodatabase.getDefaultVersionName();
+
+          // get the service feature table from the service geodatabase
+          if (serviceGeodatabase.getTable(0) != null) {
+            serviceFeatureTable = serviceGeodatabase.getTable(0);
+
+            // create a feature layer from the service feature table and add it to the map
+            featureLayer = new FeatureLayer(serviceFeatureTable);
+            map.getOperationalLayers().add(featureLayer);
+            featureLayer.addDoneLoadingListener(() -> {
+              if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+                // when the feature layer has loaded set the viewpoint and update the UI
+                mapView.setViewpointAsync(new Viewpoint(featureLayer.getFullExtent()));
+                progressIndicator.setVisible(false);
+                createVersionButton.setDisable(false);
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
+              }
+            });
+          } else {
+            new Alert(Alert.AlertType.ERROR, "Unable to get the service feature table").show();
+          }
+        } else {
+          progressIndicator.setVisible(false);
+          new Alert(Alert.AlertType.ERROR, "Service geodatabase failed to load").show();
+        }
+      });
+
+      mapView.setOnMouseClicked(event -> {
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // reset the UI
+          featureLayer.clearSelection();
+          editFeatureVBox.setDisable(true);
+
+          // select a feature
+          Point2D point = new Point2D(event.getX(), event.getY());
+          selectFeature(point);
+        }
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.SECONDARY) {
+          // if a feature is selected and the current branch version is not the default, update the feature's geometry
+          if(selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersion)){
+            Point2D secondaryClickPoint = new Point2D(event.getX(), event.getY());
+            Point mapPoint = mapView.screenToLocation(secondaryClickPoint);
```

Comment:
We could call this var 'point' as well, because there wouldn't be any clash with the variable defined above (it's in a different scope). However I see the benefit for users to give this a different name, so I wouldn't mind leaving it as is as well.

```suggestion
            Point2D point = new Point2D(event.getX(), event.getY());
            Point mapPoint = mapView.screenToLocation(point);
```

Reply:
I think I prefer just 'point' in hindsight. It's pretty blatantly in a separate block and simplifies the number of words!